feat: add obs documents (#81)
huaweicloud/huaweicloud-csi-driver: docs/obs/obs.md

# Huawei Cloud OBS CSI Driver

The OBS CSI Driver is a CSI Specification compliant driver used by Container Orchestrators to manage
the lifecycle of Huawei Cloud OBS Buckets.

## Compatibility

For sidecar version compatibility, please refer compatibility matrix for each sidecar here

> -https://kubernetes-csi.github.io/docs/sidecar-containers.html.

Reviewer: ```suggestion
https://kubernetes-csi.github.io/docs/sidecar-containers.html.
```

Author: done

## Support version

| OBS CSI Driver Version | CSI version | Kubernetes Version Tested | Features        |
|------------------------|-------------|---------------------------|-----------------|
| v0.1.0                 | v1.5.0      | v1.20 v1.21 v1.22 v1.23   | buckets resizer |

## Supported Parameters

* `acl` Optional. Specifies the ACL policy for a bucket. The predefined common policies are as follows:
`private`, `public-read`, `public-read-write`, `public-read-delivered`, `public-read-write-delivered` and
`bucket-owner-full-control`. Defaults to `private`. It is located under `parameters`.

## Deploy

### Prerequisites

- docker, kubeadm, kubelet and kubectl has been installed.

### Steps

- Create the config file

> The driver initialization depends on a [cloud config file](../../deploy/obs-csi-plugin/cloud-config).
> Make sure it's in `/etc/obs/cloud-config` on your master.

Reviewer: ```suggestion
Create the `cloud-config` file according to [cloud config file](../../deploy/obs-csi-plugin/cloud-config) in the master node.
```

Author: done

> Click to view the detailed description of the file: [cloud-config](../cloud-config.md).

Reviewer: ```suggestion
See [cloud-config](../cloud-config.md) for configurations description.
```

Author: done

- Create secret resource

```
kubectl create secret -n kube-system generic cloud-config --from-file=/etc/obs/cloud-config
```

> This should create a secret name `cloud-config` in `kube-system` namespace.

Reviewer: ```suggestion
This will create `cloud-config` secret in `kube-system` namespace.
```

Author: done

Once the secret is created, Controller Plugin and Node Plugin can be deployed using respective manifests.

- Create RBAC resources

```
kubectl apply -f https://raw.githubusercontent.com/huaweicloud/huaweicloud-csi-driver/master/deploy/obs-csi-plugin/kubernetes/rbac-csi-obs-controller.yaml
kubectl apply -f https://raw.githubusercontent.com/huaweicloud/huaweicloud-csi-driver/master/deploy/obs-csi-plugin/kubernetes/rbac-csi-obs-node.yaml
kubectl apply -f https://raw.githubusercontent.com/huaweicloud/huaweicloud-csi-driver/master/deploy/obs-csi-plugin/kubernetes/rbac-csi-obs-secret.yaml
```

- Install HuaweiCloud OBS CSI Driver

```
kubectl apply -f https://raw.githubusercontent.com/huaweicloud/huaweicloud-csi-driver/master/deploy/obs-csi-plugin/kubernetes/csi-obs-driver.yaml
kubectl apply -f https://raw.githubusercontent.com/huaweicloud/huaweicloud-csi-driver/master/deploy/obs-csi-plugin/kubernetes/csi-obs-controller.yaml
kubectl apply -f https://raw.githubusercontent.com/huaweicloud/huaweicloud-csi-driver/master/deploy/obs-csi-plugin/kubernetes/csi-obs-node.yaml
```

> - Waiting for all the pods in running

Author: done
